atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampStore.java

```diff
@@ -0,0 +1,234 @@
+                            Pair.create(CassandraTimestampUtils.INVALIDATED_VALUE.toByteArray(), currentBackupBound),
+                            CassandraTimestampUtils.BACKUP_COLUMN_NAME,
+                            Pair.create(currentBackupBound, PtBytes.EMPTY_BYTE_ARRAY)));
+            executeQueryUnchecked(client, casQueryBuffer);
+            return null;
+        });
+    }
+
+    private void checkOperationWasApplied(Optional<Long> expected, long target, CqlResult result) {
+        if (!CassandraTimestampUtils.wasOperationApplied(result)) {
+            throw constructConcurrentTimestampStoreException(expected, target, result);
+        }
+    }
+
+    private CassandraClientPool clientPool() {
+        return cassandraKeyValueService.clientPool;
+    }
+
+    private TracingQueryRunner queryRunner() {
+        return cassandraKeyValueService.getTracingQueryRunner();
+    }
+
+    private CqlResult executeQueryUnchecked(Cassandra.Client client, ByteBuffer query) {
+        try {
+            return queryRunner().run(client,
+                    AtlasDbConstants.TIMESTAMP_TABLE,
+                    () -> client.execute_cql3_query(query, Compression.NONE, ConsistencyLevel.QUORUM));
+        } catch (TException e) {
+            throw Throwables.rewrapAndThrowUncheckedException(e);
+        }
+    }
+
+    private BoundData getCurrentBoundData(Cassandra.Client client) {
+        ByteBuffer selectQuery = CassandraTimestampUtils.constructSelectFromTimestampTableQuery();
+        CqlResult existingData = executeQueryUnchecked(client, selectQuery);
+        Map<String, byte[]> columnarResults = CassandraTimestampUtils.getValuesFromSelectionResult(existingData);
+
+        return ImmutableBoundData.builder()
+                .bound(columnarResults.get(CassandraTimestampUtils.ROW_AND_COLUMN_NAME))
+                .backupBound(columnarResults.get(CassandraTimestampUtils.BACKUP_COLUMN_NAME))
+                .build();
+    }
+
+    private static ConcurrentModificationException constructConcurrentTimestampStoreException(
+            Optional<Long> expected,
+            long target,
+            CqlResult result) {
+        Optional<Long> actual = CassandraTimestampUtils.getLongValueFromApplicationResult(result);
+
+        String msg = "Unable to CAS from {} to {}."
+                + " Timestamp limit changed underneath us (limit in memory: {}, stored in DB: {}).";
+        ConcurrentModificationException except = new ConcurrentModificationException(
+                String.format(replaceBracesWithStringFormatSpecifier(msg),
+                        expected,
+                        target,
+                        expected,
+                        actual));
+        log.error(msg, expected, target, expected, actual);
+        DebugLogger.logger.error("Thread dump: {}", ThreadDumps.programmaticThreadDump());
+        throw except;
+    }
+
+    private static String replaceBracesWithStringFormatSpecifier(String msg) {
```

Comment:
Sure - I've changed `constructConcurrentTimestampStoreException` and deleted this method.